e2e/nomostest/registry_server.go

```diff
@@ -0,0 +1,358 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package nomostest
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"kpt.dev/configsync/e2e/nomostest/gitproviders"
+	"kpt.dev/configsync/e2e/nomostest/registryproviders"
+	"kpt.dev/configsync/pkg/core"
+	"kpt.dev/configsync/pkg/kinds"
+	"kpt.dev/configsync/pkg/testing/fake"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	// RegistryUsername is a contrived username used for authenticating to the in-cluster registry
+	RegistryUsername = "user"
+	// RegistryPassword is a contrived password used for authenticating to the in-cluster registry
+	RegistryPassword = "password"
+	// RegistryHTTPPort is an HTTP port surfaced on the in-cluster registry.
+	RegistryHTTPPort = 5000
+	// RegistryHTTPSPort is an HTTPS port surfaced on the in-cluster registry. The HTTPS
+	// certificate is signed using a contrived CA.
+	RegistryHTTPSPort = 5001
+	// RegistryHTTPSAuthPort is an HTTPS port surfaced on the in-cluster registry with
+	// basic authentication enabled. The HTTPS certificate is signed using a contrived CA.
+	RegistryHTTPSAuthPort = 5002
+	// TestRegistryServer is the name of the in-cluster registry Deployment and Service
+	TestRegistryServer = "test-registry-server"
+	// TestRegistryServerAuthenticated is the name of the in-cluster registry Service
+	// which requires basic auth. This endpoint can be used as a sync URL when writing
+	// tests to verify basic auth.
+	TestRegistryServerAuthenticated = "test-registry-server-auth"
+	// TestRegistryNamespace is the namespace used for the in-cluster registry
+	TestRegistryNamespace = testGitNamespace
+	nginxConfigMapName    = "nginx-cm"
+	nginxConf             = "nginx.conf"
+)
+
+// installGitServer installs the registry-server Deployment and waits for it to
+// become ready.
+func installRegistryServer(nt *NT) error {
+	objs := registryServer()
+
+	for _, o := range objs {
+		if err := nt.KubeClient.Apply(o); err != nil {
+			return fmt.Errorf("applying %v %s: %w", o.GetObjectKind().GroupVersionKind(),
+				client.ObjectKey{Name: o.GetName(), Namespace: o.GetNamespace()}, err)
+		}
+	}
+
+	return nt.Watcher.WatchForCurrentStatus(kinds.Deployment(), TestRegistryServer, TestRegistryNamespace)
+}
+
+// PushOCIImage uses the current file system state of the provided Repository
+// to build an OCI image and push it to the current RegistryProvider. The resulting
+// OCIImage object can be used to set the spec.oci.image field on the RSync.
+func (nt *NT) PushOCIImage(repository *gitproviders.Repository) (*registryproviders.OCIImage, error) {
```

Comment:
`BuildAndPushOCIImage` maybe?

Reply:
Should this maybe just be something like nt.OCIRegistry.BuildAndPush?

Reply:
Renamed. This isn't pushed into the `RegistryProvider` interface because it's intended as glue between NT and the registry provider.